public/app/features/explore/ExplorePaneContainer.tsx

```diff
@@ -46,158 +25,42 @@ const getStyles = (theme: GrafanaTheme2) => {
-        }
-      }
-
-      let queries = await ensureQueries(initialQueries, queriesDatasourceOverride); // this will return an empty array if there are no datasources
-
-      const queriesDatasourceDetails = queryDatasourceDetails(queries);
-      if (!queriesDatasourceDetails.noneHaveDatasource) {
-        if (!queryDatasourceDetails(queries).allDatasourceSame) {
-          if (config.featureToggles.exploreMixedDatasource) {
-            rootDatasourceOverride = await getDatasourceSrv().get(MIXED_DATASOURCE_NAME);
-          } else {
-            // if we have mixed queries but the mixed datasource feature is not on, change the datasource to the first query that has one
-            const changeDatasourceUid = queries.find((query) => query.datasource?.uid)!.datasource!.uid;
-            if (changeDatasourceUid) {
-              rootDatasourceOverride = changeDatasourceUid;
-              const datasource = await getDatasourceSrv().get(changeDatasourceUid);
-              const datasourceInit = await getDatasourceSrv().get(initialDatasource);
-              const newQueries = await this.props.importQueries(exploreId, queries, datasourceInit, datasource);
-              await this.props.stateSave({ replace: true });
-              queries = newQueries ?? this.props.initialQueries;
-            }
-          }
-        }
-      }
-
-      if (isFromCompactUrl) {
-        reportInteraction('grafana_explore_compact_notice');
-      }
-
-      this.props.initializeExplore({
-        exploreId,
-        datasource: rootDatasourceOverride || queries[0]?.datasource || initialDatasource,
-        queries,
-        range: initialRange,
-        containerWidth: width,
-        eventBridge: this.exploreEvents,
-        panelsState,
-        isFromCompactUrl,
-      });
-    }
-  }
-
-  componentWillUnmount() {
-    this.exploreEvents.removeAllListeners();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    this.refreshExplore(prevProps.urlQuery);
-  }
-
-  refreshExplore = (prevUrlQuery: string) => {
-    const { exploreId, urlQuery } = this.props;
-
-    // Update state from url only if it changed and only if the change wasn't initialised by redux to prevent any loops
-    if (urlQuery !== prevUrlQuery && urlQuery !== lastSavedUrl[exploreId]) {
-      this.props.refreshExplore(exploreId, urlQuery);
-    }
-  };
-
-  getRef = (el: HTMLDivElement) => {
-    this.el = el;
-  };
-
-  render() {
-    const { theme, exploreId, initialized, eventBus } = this.props;
-    const styles = getStyles(theme);
-    return (
-      <div className={styles.explore} ref={this.getRef} data-testid={selectors.pages.Explore.General.container}>
-        {initialized && <Explore exploreId={exploreId} eventBus={eventBus} />}
-      </div>
-    );
-  }
+/*
+  Connected components subscribe to the store before function components (using hooks) and can react to store changes. Thus, this connector function is called before the parent component (ExplorePage) is rerendered.
```

Comment:
Thanks for writing it down 👍  It'd be worth to have a GitHub Issue to look deeper into this. Also, we usually use  `/**` comment blocks with `*` on each line.

Reply:
comments starting with`/**` are used to document functions with jsdoc (so that it shows up when doing autocomplete for instance). this is just a multiline comment explaining why that was needed. how would having a jsdoc block help here?

Reply:
> how would having a jsdoc block help here?

It looks relevant to the component so works as a function doc to me. I'm fine with both, it's more about style convention - I think for multiline comments we use `//` on each line, and I haven't seen much `/* */` for multi-lines. It's more just a personal preference, as there's no linting rule to enforce anything.